src/transformStateWithClones.js

```diff
@@ -1,13 +1,39 @@
 'use strict';
 
 /**
- * @param {Object} state
+ * @param {Object}  currentState
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      currentState = {};
+      states.push({ ...currentState });
+    }
+
+    if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+      states.push({ ...currentState });
+    }
+
+    if (action.type === 'removeProperties') {
```

Comment:
Consider using a switch statement instead of multiple if statements for handling different action types. This will make the code cleaner and more maintainable. Refer to the checklist item: 'Use switch statement if you have limited amount of conditions.'